google/cloud/internal/curl_rest_client.cc

```diff
@@ -52,7 +52,9 @@ Status MakeRequestWithPayload(
     }
     encoded_payload = impl.MakeEscapedString(concatenated_payload);
     if (!encoded_payload.empty()) {
```

Comment:
can we get rid of this if altogether? does `encoded_payload.empty() => encoded_payload.size() == 0`?